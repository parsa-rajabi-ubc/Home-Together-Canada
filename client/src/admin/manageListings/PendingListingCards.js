/**
 *
 * @Author:     Parsa Rajabi
 * @Created:    2021.3.20
 *
 * @Description: Pending Listing Cards
 *
 */

import React from "react";
import PropTypes from "prop-types";
import BusinessListingCard from "../../searchServicesClassifieds/listingCards/BusinessListingCard";
import {AiFillCheckCircle} from "react-icons/ai";
import {AiFillCloseCircle} from "react-icons/ai";
import Modal from 'react-modal';


const ADMIN_TEXT = {
    TITLE: "Pending Listings",
    INFO: "Use the icons to the right of each listing to approve or reject them",
    CONFIRM_APPROVE: "Are you sure you want to approve listing #",
    CONFIRM_DECLINE: "Are you sure you want to reject listing #"
}

Modal.setAppElement('body')
const customModalStyles = {
    content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        padding: '60px',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
        backgroundColor: 'white',
        borderRadius: '10px',
    }
};

function PendingListingCards(props) {
    const {
        onSubmit,
        listingID,
        setListingID,
        listingStatus,
        setListingStatus,
        pendingListings,
    } = props;

    const [IsModalOpen, setIsModalOpen] = React.useState(false);

    function openModal() {
        setIsModalOpen(true);
    }

    function closeModal() {
        setIsModalOpen(false);
    }

    function approveListing() {
        closeModal();
        onSubmit();
    }

    const onClickDecision = (decision, listingID) => {
        setListingID(listingID);
        setListingStatus(decision);
        openModal();
    }

    const listingCards = pendingListings.map(
        (listing) =>
            <div className={"flex"} key={listing.id}>
                <section className={"w-full"}>
                    <BusinessListingCard
                        logo={listing.id.toString()}
                        title={listing.title}
                        businessName={listing.businessName}
                        shortDescription={listing.shortDescription}
                        datePosted={listing.createdAt}/>
                </section>
                <section className={"my-8"}>
                    <button
                        className={"focus:outline-none"} type="button"
                        onClick={e => onClickDecision(true, listing.id, e)}>
                        <AiFillCheckCircle
                            id={"approve"}
                            className={"text-green-600 hover:text-green-700 hover:cursor-pointer align-top"}
                            size="40"
                        />
                    </button>
                    <button className={"focus:outline-none"} type="button"
                            onClick={e => onClickDecision(false, listing.id, e)}>
                        <AiFillCloseCircle
                            className={"text-red-600 hover:text-red-700 hover:cursor-pointer align-bottom"}
                            size="40"/>
                    </button>
                </section>
            </div>
    );

    return (
        <div>
            <h3 className={"account-summary-info-header"}> {ADMIN_TEXT.TITLE} </h3>
            <p className="account-summary-info-text"> {ADMIN_TEXT.INFO} </p>

            <div className={"ml-10 my-3"}>{listingCards}</div>
            <Modal
                isOpen={IsModalOpen}
                onRequestClose={closeModal}
                style={customModalStyles}
                contentLabel="Confirm Decision for Listing"
            >
                <h1 className={"text-gray-800 h1"}>{listingStatus ? ADMIN_TEXT.CONFIRM_APPROVE : ADMIN_TEXT.CONFIRM_DECLINE}{listingID + "?"} </h1>
                <section className={"flex pt-10 space-x-10"}>
                    <button type={"button"} className="py-2 w-1/2 bg-gray-500 btn" onClick={closeModal}>
                        Cancel
                    </button>
                    <button type={"submit"}
                            className={`${listingStatus ? "btn-green" : "btn-red"} btn w-1/2 py-2`}
                            onClick={approveListing}
                            value={listingStatus ? "Approve" : "Reject"}
                    >
                        {listingStatus ? "Approve" : "Reject"}
                    </button>
                </section>
            </Modal>
        </div>
    );
}

PendingListingCards.propTypes = {
    onSubmit: PropTypes.func.isRequired,
    listingID: PropTypes.number,
    setListingID: PropTypes.func.isRequired,
    listingStatus: PropTypes.bool,
    setListingStatus: PropTypes.func.isRequired,
    pendingListings: PropTypes.array.isRequired
};

export default PendingListingCards;