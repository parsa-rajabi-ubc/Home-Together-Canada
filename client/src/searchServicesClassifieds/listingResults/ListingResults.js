/**
 *
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.19
 *
 * @Description: Listing Results Component
 *
 */

import React, {useContext, useEffect, useState} from 'react';
import {Link} from "react-router-dom";
import Paginate from "../../common/forms/Paginate";
import {listingContext, PAGE_NAMES} from "../SearchListingContainer";
import BusinessListingCard from "../listingCards/BusinessListingCard";
import {mockBusinessListings, mockMemberListings} from "../../mockData/MockListingData";
import MemberListingCard from "../listingCards/MemberListingCard";

const NUM_RESULTS = 7;

function ListingResults() {

    const listingPage = useContext(listingContext);

    const businessCards = mockBusinessListings.map(
        (business) =>
            <Link to={`/${listingPage}/${business.id}`} key={business.id}>
                <BusinessListingCard
                    logo={business.logo}
                    title={business.title}
                    businessName={business.businessName}
                    shortDescription={business.shortDescription}
                    datePosted={business.datePosted}/>
            </Link>
    )

    const memberListingCards = mockMemberListings.map(
        (member) =>
            <Link to={`/${listingPage}/${member.id}`} key={member.id}>
                <MemberListingCard
                    title={member.title}
                    monthlyCost={member.monthlyCost}
                    petFriendly={member.petFriendly}
                    smokeFriendly={member.smokeFriendly}
                    shortDescription={member.shortDescription}
                    datePosted={member.datePosted}/>
            </Link>
    )

    const [listingCards, setListingCards] = useState((listingPage === PAGE_NAMES.SERVICES) ? businessCards.concat(memberListingCards) : businessCards)

    useEffect(() => {
        setListingCards((listingPage === PAGE_NAMES.SERVICES) ? businessCards.concat(memberListingCards) : businessCards)
    }, [listingPage]);


    return (
        <div>
            <Paginate data={listingCards} resultsPerPage={NUM_RESULTS}/>
        </div>
    );
}


export default ListingResults;