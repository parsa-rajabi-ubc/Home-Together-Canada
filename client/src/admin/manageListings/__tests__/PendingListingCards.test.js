/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.3.21
 *
 * @Description: Pending Listing Cards Test
 *
 */

import React from 'react';
import renderer from 'react-test-renderer'
import PendingListingCards from "../PendingListingCards";
import ListingResults from "../../../searchServicesClassifieds/listingResults/ListingResults";
import {BrowserRouter as Router} from "react-router-dom";

describe('PendingListingCards', () => {
    describe('Container test', () => {
        it('should match snapshot test regardless of the props (approved listing)', () => {
            //given
            const props = {
                listingID: 2,
                setListingID: jest.fn(),
                onSubmit: jest.fn(),
                setListingStatus: jest.fn(),
                listingStatus: true,
                pendingListings: [{
                    id: 20,
                    title: "Forest Green Co-Living",
                    shortDescription: "Easy, family friendly, & communal living",
                    createdAt: "2021-03-21T22:16:06.000Z",
                    business: {
                        businessName: "Bob's Bricks",
                    }
                }]
            }
            //when
            const component = renderer.create(<Router><PendingListingCards {...props}/></Router>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
        it('should match snapshot test regardless of the props (rejected listing)', () => {
            //given
            const props = {
                listingID: 3,
                setListingID: jest.fn(),
                onSubmit: jest.fn(),
                setListingStatus: jest.fn(),
                listingStatus: false,
                pendingListings: [{
                    id: 20,
                    title: "Forest Green Co-Living",
                    shortDescription: "Easy, family friendly, & communal living",
                    createdAt: "2021-03-21T22:16:06.000Z",
                    business: {
                        businessName: "Bob's Bricks",
                    }
                }]
            }
            //when
            const component = renderer.create(<Router><PendingListingCards {...props}/></Router>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });
});