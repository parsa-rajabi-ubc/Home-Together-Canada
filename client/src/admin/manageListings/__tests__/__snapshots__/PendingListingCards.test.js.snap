// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingListingCards Container test should match snapshot test regardless of the props (approved listing) 1`] = `
<div>
  <h3
    className="account-summary-info-header"
  >
     
    Pending Listings
     
  </h3>
  <p
    className="account-summary-info-text"
  >
     
    Use the icons to the right of each listing to approve or reject them
     
  </p>
  <div
    className="ml-10 my-3"
  >
    <div>
      <div
        className="flex"
      >
        <section
          className="w-full"
        >
          <section
            className="card-container mt-0"
          >
            <img
              className="float-left w-24 h-24 mx-4 "
              src="HTC_Logo.jpg"
            />
            <div
              className="inline align-middle "
            >
              <label
                className="font-semibold justify-between"
              >
                 
                Forest Green Co-Living
                 
              </label>
              <section
                className="inline float-right pr-6 justify-end"
              >
                <time
                  dateTime={"2021-03-21T22:16:06.000Z"}
                >
                  Mar 21, 2021
                </time>
              </section>
            </div>
            <div
              className="mx-6 inline items-center justify-between leading-tight font-light"
            >
              <section
                className="mb-3"
              >
                 
                Bob's Bricks
                 
              </section>
              <section>
                 
                Easy, family friendly, & communal living
                 
              </section>
            </div>
          </section>
        </section>
        <section
          className="my-8"
        >
          <button
            className="focus:outline-none"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="text-green-600 hover:text-green-700 hover:cursor-pointer align-top"
              fill="currentColor"
              height="40"
              id="approve"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
              />
            </svg>
          </button>
          <button
            className="focus:outline-none"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="text-red-600 hover:text-red-700 hover:cursor-pointer align-bottom"
              fill="currentColor"
              height="40"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </button>
        </section>
      </div>
      <ul
        className="flex m-auto w-2/3"
      >
        <li
          className="next-prev-text disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            className="next-prev-link"
            onClick={[Function]}
            onKeyPress={[Function]}
            rel="prev"
            role="button"
            tabIndex="0"
          >
            Previous
          </a>
        </li>
        <li
          className="selected"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            className="page-numbers current-page"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            1
          </a>
        </li>
        <li
          className="next-prev-text disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Next page"
            className="next-prev-link"
            onClick={[Function]}
            onKeyPress={[Function]}
            rel="next"
            role="button"
            tabIndex="0"
          >
            Next
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`PendingListingCards Container test should match snapshot test regardless of the props (rejected listing) 1`] = `
<div>
  <h3
    className="account-summary-info-header"
  >
     
    Pending Listings
     
  </h3>
  <p
    className="account-summary-info-text"
  >
     
    Use the icons to the right of each listing to approve or reject them
     
  </p>
  <div
    className="ml-10 my-3"
  >
    <div>
      <div
        className="flex"
      >
        <section
          className="w-full"
        >
          <section
            className="card-container mt-0"
          >
            <img
              className="float-left w-24 h-24 mx-4 "
              src="HTC_Logo.jpg"
            />
            <div
              className="inline align-middle "
            >
              <label
                className="font-semibold justify-between"
              >
                 
                Forest Green Co-Living
                 
              </label>
              <section
                className="inline float-right pr-6 justify-end"
              >
                <time
                  dateTime={"2021-03-21T22:16:06.000Z"}
                >
                  Mar 21, 2021
                </time>
              </section>
            </div>
            <div
              className="mx-6 inline items-center justify-between leading-tight font-light"
            >
              <section
                className="mb-3"
              >
                 
                Bob's Bricks
                 
              </section>
              <section>
                 
                Easy, family friendly, & communal living
                 
              </section>
            </div>
          </section>
        </section>
        <section
          className="my-8"
        >
          <button
            className="focus:outline-none"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="text-green-600 hover:text-green-700 hover:cursor-pointer align-top"
              fill="currentColor"
              height="40"
              id="approve"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
              />
            </svg>
          </button>
          <button
            className="focus:outline-none"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="text-red-600 hover:text-red-700 hover:cursor-pointer align-bottom"
              fill="currentColor"
              height="40"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 1024 1024"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </button>
        </section>
      </div>
      <ul
        className="flex m-auto w-2/3"
      >
        <li
          className="next-prev-text disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            className="next-prev-link"
            onClick={[Function]}
            onKeyPress={[Function]}
            rel="prev"
            role="button"
            tabIndex="0"
          >
            Previous
          </a>
        </li>
        <li
          className="selected"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            className="page-numbers current-page"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            1
          </a>
        </li>
        <li
          className="next-prev-text disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Next page"
            className="next-prev-link"
            onClick={[Function]}
            onKeyPress={[Function]}
            rel="next"
            role="button"
            tabIndex="0"
          >
            Next
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
